spec/requests/api/v0.1/post_hesa_trainees_spec.rb

```diff
@@ -202,11 +208,78 @@
     it "sets the provider_trainee_id" do
       expect(Trainee.last.provider_trainee_id).to eq("99157234/2/01")
     end
+
+    context "when read only attributes are been submitted" do
+      let(:trn) { "567899" }
+      let(:ethnic_group) { "mixed_ethnic_group" }
+      let(:ethnic_background) { "Another Mixed background" }
+
+      context "when the ethnicity is provided" do
+        let(:params) do
+          {
+            data: data.merge(
+              trn:,
+              ethnicity:,
+              ethnic_group:,
+              ethnic_background:,
+            ),
+          }
+        end
+
+        let(:ethnicity) { "899" }
+
+        it "sets the ethnic attributes based on ethnicity" do
+          expect(response).to have_http_status(:created)
```

Comment:
Hi @stevehook

I understand your concern that from a user's perspective they may expect some kind of validation logic that they tried to set attributes to the Trainee that are not meant to be set manually.

A pattern that I have seen in APIs is to just ignore unsupported attributes as the expectation is for the users to follow the schema that we have defined for them 😄 

Having said that, what you suggested might be useful to add in a further iteration and would probably have to go through all read only attributes in the API and come into an agreement on how we feedback to the end user.

Reply:
Good points on both approaches to this. For now (this PR) I think I'm happy with ignoring read-only attributes, in the same way as we ignore unrecognised attributes sent in requests.

However, I think it would be good for us to provide feedback on read-only attributes in the future, either from existing POST/PUT endpoints, or as part of specific validation endpoints if we implement those.

I'll add a ticket to the backlog to capture this.